File: src/main/java/com/sequenceiq/periscope/monitor/evaluator/EvaluatorExecutor.java
package com.sequenceiq.periscope.monitor.evaluator;

import java.util.Map;

import org.springframework.context.ApplicationEventPublisher;
import org.springframework.context.ApplicationEventPublisherAware;

public interface EvaluatorExecutor extends ApplicationEventPublisher, ApplicationEventPublisherAware, Runnable {

    void setContext(Map<String, Object> context);

}

File: autoscale/src/main/java/com/sequenceiq/periscope/monitor/evaluator/EvaluatorExecutor.kt
package com.sequenceiq.periscope.monitor.evaluator

import org.springframework.context.ApplicationEventPublisher
import org.springframework.context.ApplicationEventPublisherAware

interface EvaluatorExecutor : ApplicationEventPublisher, ApplicationEventPublisherAware, Runnable {

    fun setContext(context: Map<String, Any>)

}
